From 5bc522136fd7f4cc19f9b0b903eeb28d54044475 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 14 Mar 2019 03:55:47 -0700
Subject: [PATCH 0437/1852] dt-bindings: media: xilinx: demosaic: Make max
 width, height mandatory

commit 70f6c09db630a3a38f22fb14e0099bae6ff72231 from
https://github.com/Xilinx/linux-xlnx.git

This patch makes the xlnx,max-width and xlnx,max-height dt properties
mandatory and breaks backwards compatibility.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/media/xilinx/xlnx,v-demosaic.txt       | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/Documentation/devicetree/bindings/media/xilinx/xlnx,v-demosaic.txt b/Documentation/devicetree/bindings/media/xilinx/xlnx,v-demosaic.txt
index e43ce1adf8a1..9b3aff413e0e 100644
--- a/Documentation/devicetree/bindings/media/xilinx/xlnx,v-demosaic.txt
+++ b/Documentation/devicetree/bindings/media/xilinx/xlnx,v-demosaic.txt
@@ -13,6 +13,10 @@ Required properties:
 
 - clocks: Reference to the AXI Streaming clock feeding the Demosaic ap_clk.
 
+- xlnx,max-height: Maximum number of lines. Valid range is 64 to 4320.
+
+- xlnx,max-width: Maximum number of pixels in a line. Valid range is 64 to 8192.
+
 - reset-gpios: Specifier for GPIO that asserts Demosaic IP (AP_RST_N) reset.
 
 - ports: Video ports, using the DT bindings defined in ../video-interfaces.txt.
@@ -25,17 +29,14 @@ Required port properties:
 
 - xlnx,video-width: Video width as defined in video.txt
 
-Optional properties:
-
-- xlnx,max-height: Maximum number of lines.
-- xlnx,max-width: Maximum number of pixels in a line.
-
-Example :
+Example:
 	demosaic_1: demosaic@a00b0000 {
 		compatible = "xlnx,v-demosaic";
 		reg = <0x0 0xa00b0000 0x0 0x10000>;
 		clocks = <&vid_stream_clk>;
 		reset-gpios = <&gpio 87 1>;
+		xlnx,max-width = <3840>;
+		xlnx,max-height = <2160>;
 		ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
-- 
2.31.1

